### PR TITLE
Experimental environment for emulating a device shell

### PR DESCRIPTION
**Pull request by estenberg**
Confirmed to work for doing `apt upgrade` with Raspberry Pi 3.

### CODE REVIEW

**Comment by estenberg** [comment]
If this is generally useful and works we can generalize it (remove the mender-artifact generation) and put it in its own repo.

File: device-image-shell/docker-device-image-shell [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+CURRENT_USER=$(id -u -n)
+CURRENT_GROUP=$(id -g -n)
+
+echo "Changing ownership of Mender Artifact (may ask you to authenticate)"
+sudo chown $CURRENT_USER:$CURRENT_GROUP $OUTPUT_DIR/$ARTIFACT_NAME.mender
+
+echo "Image generation complete!"
+echo "The new root file system is at:\n\t$OUTPUT_DIR/$ROOTFS_OUTPUT_FILE_NAME"
+echo "The new Mender Artifact you can upload to your Mender server to deploy to your devices is at:\
+         \n\t$OUTPUT_DIR/$ARTIFACT_NAME.mender"
```
[the remark this answers is not in the document]
Comment (apodogrocki) [reply device-image-shell/docker-device-image-shell]: `echo -e` to enable interpretation of backslash escapes

[the remark this answers is not in the document]
Comment (estenberg) [reply device-image-shell/docker-device-image-shell]: Thanks! Replaced with echo -e.

File: device-image-shell/docker-device-image-shell [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+       --mount type=bind,source=$OUTPUT_DIR,target=/root_images \
+       $DOCKER_IMAGE_NAME $ROOTFS_OUTPUT_FILE_NAME $ARTIFACT_NAME $DEVICE_TYPE
+
+
+# Output Artifact gets root owner and group, change to current logged in
+CURRENT_USER=$(id -u -n)
+CURRENT_GROUP=$(id -g -n)
+
+echo "Changing ownership of Mender Artifact (may ask you to authenticate)"
+sudo chown $CURRENT_USER:$CURRENT_GROUP $OUTPUT_DIR/$ARTIFACT_NAME.mender
```
[the remark this answers is not in the document]
Comment (estenberg) [reply device-image-shell/docker-device-image-shell]: Thanks, though the chown did not have needed perms without sudo for me.
The Mender Artifact is owned by root (it is created inside container, which is run privileged).

If you have ideas I would be happy to get rid of this.

**Comment by estenberg** [comment]
Thanks for testing Mirza:
* Added echo -e
* Added note that commands are run as root

It makes sense to run "mount -t devpts devpts /dev/pts", however I am not sure how to run commands after it enters chroot automatically - it is up to the user then?

**Pull request merged**